src/activities/content/ContentHtmlFileEntity.js

```diff
@@ -0,0 +1,133 @@
+import { Actions, Rels } from '../../hypermedia-constants.js';
+import { performSirenAction } from '../../es6/SirenAction.js';
+import ContentHelperFunctions from './ContentHelperFunctions.js';
+import { ContentWorkingCopyEntity } from './ContentWorkingCopyEntity.js';
+
+/**
+ * ContentHtmlFileEntity class representation of a d2l content-htmlFile entity.
+ */
+export class ContentHtmlFileEntity extends ContentWorkingCopyEntity {
+
+	/**
+	 * @returns {string|null} Description html of the content-htmlFile item
+	 */
+	descriptionRichText() {
+		const descriptionSubEntity = ContentHelperFunctions.getDescriptionSubEntity(this._entity);
+		if (!descriptionSubEntity) {
+			return null;
+		}
+		return descriptionSubEntity.properties.html || '';
+	}
+
+	/**
+	 * @returns {string|null} Description text of the content-htmlFile item
+	 */
+	descriptionText() {
+		const descriptionSubEntity = ContentHelperFunctions.getDescriptionSubEntity(this._entity);
+		if (!descriptionSubEntity) {
+			return null;
+		}
+		return descriptionSubEntity.properties.text || '';
+	}
+
+	/**
+	 * @returns {string} Title of the content-htmlFile item
+	 */
+	title() {
+		return this._entity && this._entity.properties && this._entity.properties.title;

```

Comment:
question about this one here. looking at `descriptionText()` and `descriptionRichText()`, the comment suggests it can return `{string|null}` which makes sense looking at what can happen, but looking at this line here and the fact that it suggests it will always return a string, something doesn't look quite right.

There is checking at each step suggesting that this could short circuit, but there's no fallback or anything. So if `title` isn't there for whatever reason, we're going to return `undefined`. So I would suggest either changing the return type description, or removing the null checking if we're confident it will exist. Or add a fallback empty string like the others.